File: C++/C++Basic/08.for-Loop-Exercise/06.oscars.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    string actorName;
    getline (cin, actorName);

    double academyPoint;
    cin >> academyPoint;
    
    int juryCount;
    cin >> juryCount;

    cin.ignore();

    cout.setf(ios::fixed);
    cout.precision(1);

    for (int i = 1; i <= juryCount; i++)
    {
        string juryName;
        getline(cin, juryName);

        double juryPoints;
        cin >> juryPoints;
        cin.ignore();

        academyPoint += juryName.length() * juryPoints / 2.0;

        if (academyPoint > 1250.5)
        {
            cout << "Congratulations, " << actorName << " got a nominee for leading role with " << academyPoint << "!" << endl;
            break;
        }

    }

    if (academyPoint <= 1250.5)
    {
        cout << "Sorry, " << actorName << " you need " << abs(1250.5 - academyPoint) << " more!" << endl;
    }
    

    return 0;
}

File: Python/Basic/08.For-Loop-Exersice/06.oscars.py
actor_name = input()
academy_point = float(input())
jury_count = int(input())

for _ in range(jury_count):
    jury_name = input()
    points = float(input())

    academy_point += len(jury_name) * points / 2
    
    if academy_point >= 1250.5:
        print(f"Congratulations, {actor_name} got a nominee for leading role with {academy_point:.1f}!")
        break
else:
    print(f"Sorry, {actor_name} you need {abs(academy_point - 1250.5):.1f} more!")



